### DISCUSSION TITLE
VPN tunnel stops working after some time - received message ID X, expected X+1, ignored

### DISCUSSION THREAD

**Discussion by xnr13** [strongswan/strongswan#2525] [General]
On cold boot VPN tunnels are established and work fine. After some network changes (e.g. roadwarrior goes to sleep and the wakes up), the tunnels first gets re-established (see 12:12:XX time block), works for a few minutes and then loses traffic (see 12:15:XX time block). The connection then indefinitely retries with messages like "received message ID X, expected X+1, ignored" in the log. What can i do? I googled, found [this](https://wiki.strongswan.org/issues/816), not sure if it is relevant.

[charon_debug.log](https://github.com/user-attachments/files/17647217/charon_debug.log)
[swanctl -l (2).txt](https://github.com/user-attachments/files/17647218/swanctl.-l.2.txt)
[swanctl -L.txt](https://github.com/user-attachments/files/17647219/swanctl.-L.txt)
[swanctl.conf.txt](https://github.com/user-attachments/files/17647223/swanctl.conf.txt)
[iptables-save.txt](https://github.com/user-attachments/files/17647166/iptables-save.txt)

**Comment by tobiasbrunner** [comment]
That sounds like your peer sends a different message (hash doesn't match) with the same message ID as the one already received and responded to (but apparently the response did not arrive there).

**Reply by keskinadem** [reply]
What is the solution for this case?

**Reply by tobiasbrunner** [reply]
There is none, it's most likely a bug in the other implementation.

**Reply by xnr13** [reply]
> There is none, it's most likely a bug in the other implementation.

Would you have an idea of a workaround configuration on the gateway?

**Reply by keskinadem** [reply]
Did you try to solve by changing some configuration? Maybe try longer timeout or something IDK

**Comment by keskinadem** [comment]
@xnr13 How did you solve this issue?

**Reply by xnr13** [reply]
> @xnr13 How did you solve this issue?

I did not. What's your setup? Mine is W10 roadwarriors and a Linux gateway.

**Reply by keskinadem** [reply]
Both side linux.
